src/core/server/core_usage_data/core_usage_stats_client.ts

```diff
@@ -88,67 +105,124 @@ export class CoreUsageStatsClient {
+    await this.updateUsageStats(counterFieldNames, IMPORT_STATS_PREFIX, options);
   }
 
-  public async incrementSavedObjectsResolveImportErrors({
-    headers,
-    createNewCopies,
-  }: IncrementSavedObjectsResolveImportErrorsOptions) {
-    const isKibanaRequest = getIsKibanaRequest(headers);
-    const counterFieldNames = [
-      'total',
-      `kibanaRequest.${isKibanaRequest ? 'yes' : 'no'}`,
-      `createNewCopiesEnabled.${createNewCopies ? 'yes' : 'no'}`,
-    ];
-    await this.updateUsageStats(counterFieldNames, RESOLVE_IMPORT_STATS_PREFIX);
+  public async incrementSavedObjectsResolveImportErrors(
+    options: IncrementSavedObjectsResolveImportErrorsOptions
+  ) {
+    const { createNewCopies } = options;
+    const counterFieldNames = [`createNewCopiesEnabled.${createNewCopies ? 'yes' : 'no'}`];
+    await this.updateUsageStats(counterFieldNames, RESOLVE_IMPORT_STATS_PREFIX, options);
   }
 
-  public async incrementSavedObjectsExport({
-    headers,
-    types,
-    supportedTypes,
-  }: IncrementSavedObjectsExportOptions) {
-    const isKibanaRequest = getIsKibanaRequest(headers);
+  public async incrementSavedObjectsExport(options: IncrementSavedObjectsExportOptions) {
+    const { types, supportedTypes } = options;
     const isAllTypesSelected = !!types && supportedTypes.every((x) => types.includes(x));
-    const counterFieldNames = [
-      'total',
-      `kibanaRequest.${isKibanaRequest ? 'yes' : 'no'}`,
-      `allTypesSelected.${isAllTypesSelected ? 'yes' : 'no'}`,
-    ];
-    await this.updateUsageStats(counterFieldNames, EXPORT_STATS_PREFIX);
+    const counterFieldNames = [`allTypesSelected.${isAllTypesSelected ? 'yes' : 'no'}`];
+    await this.updateUsageStats(counterFieldNames, EXPORT_STATS_PREFIX, options);
   }
 
-  private async updateUsageStats(counterFieldNames: string[], prefix: string) {
+  private async updateUsageStats(
+    counterFieldNames: string[],
+    prefix: string,
+    { request }: BaseIncrementOptions
+  ) {
     const options = { refresh: false };
     try {
       const repository = await this.repositoryPromise;
+      const fields = [...this.getCommonFieldsToIncrement(request), ...counterFieldNames];
       await repository.incrementCounter(
         CORE_USAGE_STATS_TYPE,
         CORE_USAGE_STATS_ID,
-        counterFieldNames.map((x) => `${prefix}.${x}`),
+        fields.map((x) => `${prefix}.${x}`),
         options
       );
     } catch (err) {
       // do nothing
     }
   }
+
+  private getIsDefaultNamespace(request: KibanaRequest) {
+    const requestBasePath = this.basePath.get(request); // obtain the original request basePath, as it may have been modified by a request interceptor
+    const pathToCheck = this.basePath.remove(requestBasePath); // remove the server basePath from the request basePath
+    const matchResult = pathToCheck.match(SPACE_CONTEXT_REGEX); // Look for `/s/space-url-context` in the base path
```

Comment:
🙈 , but I don't see any other easy option if we need this to be in core...

Reply:
😬

Reply:
I haven't tested this (sorry...not at my laptop at the moment), but could we infer this by checking to see if the `requestBasePath` === `this.basePath.serverBasePath`? If they match, then we'd be in the default space. If the request has a modified base path, then we'd be in the non-default space. This wouldn't account for users who explicitly target the default space via `/s/default`, but I'd expect (with no evidence) that this is uncommon.

Feel free to ignore this, just pitching a potential alternative

Reply:
I tested it and it does seem to work. I'd be hesitant to use this without accounting for `/s/default`, though, and we have an explicit check for that it doesn't really simplify our code. At least with the current approach, it's the same in Core and the Spaces plugin, so if it ever needs to change for some reason, it'll be more evident that the same change needs to be made in both places.

I'll leave it as-is for now and if we decide to revisit in #85869 we can do so.